src/Service/RevisionSaver.php

```diff
@@ -36,6 +37,14 @@ public function __construct( MediawikiApi $api ) {
 	 * @returns bool success
 	 */
 	public function save( Revision $revision, EditInfo $editInfo = null ) {
+		if( is_null( $editInfo ) ) {
+			if( is_null( $revision->getEditInfo() )) {
+				$editInfo = new EditInfo();
+			} else {
+				$editInfo = $revision->getEditInfo();
+			}
+		}
```

Comment:
This can thus be done in a one liner!

$editInfo = $editInfo ? $editInfo : $revision->getEditInfo();